### PR TITLE
Assign assessors

### PR DESCRIPTION
**Pull request by jaleszek**
Bulk assessors assignment. As Lead Assessor I want to assign primary/secondary assessors for  multiple applications at once.

### CODE REVIEW

File: app/assets/javascripts/admin/form_answers.js.coffee [hunk 1]
```diff
@@ -55,6 +72,21 @@ ready = ->
       $(this).closest(".panel-body").removeClass("audit-cert-changes-made")
       $(this).closest(".panel-body").addClass("audit-cert-no-change")
 
+  $(document).on "submit", "#new_assessor_assignment_collection", (e) ->
+    form = $(this)
+    ids = ""
+    $(".form-answer-check:checked").each ->
+      ids += ($(@).val() + ",")
```
Comment (matthewford) [inline app/assets/javascripts/admin/form_answers.js.coffee]: @ must not be used stand alone

File: app/controllers/concerns/assessor_assignment_context.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module AssessorAssignmentContext
+  def update
```
Comment (matthewford) [inline app/controllers/concerns/assessor_assignment_context.rb]: Assignment Branch Condition size for update is too high. [15.07/15]

File: app/models/assessor.rb [hunk 3]
```diff
@@ -65,10 +65,32 @@ def lead?(form_answer)
...
+  def assignable?(form_answer)
+    lead?(form_answer) || regular?(form_answer)
+  end
+
   def lead_or_assigned?(form_answer)
     lead?(form_answer) || assigned?(form_answer)
   end
 
+  def categories_as_lead
+    categories.select { |k, v| v == "lead" }.keys
```
Comment (matthewford) [inline app/models/assessor.rb]: Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.

File: app/models/assessor_assignment.rb [hunk 4]
```diff
@@ -22,6 +22,9 @@ class AssessorAssignment < ActiveRecord::Base
     validate :submitted_at_immutability
     validate :assessor_existence
     validate :assessor_assignment_to_category
+    validates :assessor_id,
```
Comment (matthewford) [inline app/models/assessor_assignment.rb]: Inconsistent indentation detected.

File: app/models/assessor_assignment_collection.rb [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      prim.save || add_error(prim.errors.to_a)
+    end
+
+    if secondary_assignable?(fa)
+      sec.assessor = secondary_assessor
+      sec.save || add_error(sec.errors.to_a)
+    end
+  end
+
+  def handle_double_assignment(fa, prim, sec)
```
Comment (matthewford) [inline app/models/assessor_assignment_collection.rb]: Assignment Branch Condition size for handle_double_assignment is too high. [17.09/15]

File: spec/features/assessor/bulk_assessors_assignment_spec.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
+require "rails_helper"
+include Warden::Test::Helpers
+
+Warden.test_mode!
+
+describe "Assessor submits appraisal form", %(
+  As Assessor
+  I want to be able to edit, submit the appraisal form.
+) do
+
```
Comment (matthewford) [inline spec/features/assessor/bulk_assessors_assignment_spec.rb]: Extra empty line detected at block body beginning.

File: app/assets/javascripts/admin/form_answers.js.coffee (hunk 1 above)
Comment (matthewford) [inline app/assets/javascripts/admin/form_answers.js.coffee]: @ must not be used stand alone

File: app/controllers/concerns/assessor_assignment_context.rb (hunk 2 above)
Comment (matthewford) [inline app/controllers/concerns/assessor_assignment_context.rb]: Assignment Branch Condition size for update is too high. [15.07/15]

File: app/models/assessor_assignment.rb (hunk 4 above)
Comment (matthewford) [inline app/models/assessor_assignment.rb]: Inconsistent indentation detected.

File: app/models/assessor_assignment_collection.rb (hunk 5 above)
Comment (matthewford) [inline app/models/assessor_assignment_collection.rb]: Assignment Branch Condition size for handle_double_assignment is too high. [17.09/15]

**Comment by jaleszek** [comment]
@RuslanHamidullin can you take a look please? ; )

**Comment by RuslanHamidullin** [comment]
Good job @jaleszek :+1:

**Pull request merged**